lib/DxilContainer/DxilContainerAssembler.cpp

```diff
@@ -1556,10 +1556,6 @@ void hlsl::SerializeDxilContainerForModule(DxilModule *pModule,
 
   // If metadata was stripped, re-serialize the module.
   CComPtr<AbstractMemoryStream> pInputProgramStream = pModuleBitcode;
```

Comment:
It would help making this function understandable to rename variables a bit:
- bModuleDirty > bModuleSignatureMetadataStripped (because later you modify the module and do not mark it dirty)
- pInputProgramStream > pDebugPartBitcode
- bModuleStripped > bModuleDebugInfoStripped (because it doesn't include signature metadata stripping)
- pProgramStream > pFinalDxilBitcode

Or better yet, break this into a few functions:

SerializeDxilContainerForModule
{
DxilContainerWriter_impl writer;
WriteSignatureParts(writer, pModule, pModuleBitcode) : bool (module modified)
// update pModuleBitcode if needed
WriteDebugInfoParts(writer, pModule, pModuleBitcode, DebugName, flags) : bool (module modified)
// update pModuleBitcode if needed
// Write DXIL part
}

Reply:
Did small rename in commit 3. We could do big change later.